tests/Tokenizer/Transformer/NameQualifiedTransformerTest.php

```
@@ -75,88 +75,84 @@ public static function provideProcessCases(): iterable
             ],
         ];
 
-        if (\defined('T_NAME_RELATIVE')) { // @TODO: drop condition when PHP 8.0+ is required
-            yield 'relative 1' => [
-                [
-                    new Token([T_OPEN_TAG, "<?php\n"]),
-                    new Token([T_NAMESPACE, 'namespace']),
-                    new Token([T_NS_SEPARATOR, '\\']),
-                    new Token([T_STRING, 'Transformer']),
-                    new Token(';'),
-                ],
-                [
-                    new Token([T_OPEN_TAG, "<?php\n"]),
-                    new Token([T_NAME_RELATIVE, 'namespace\Transformer']),
-                    new Token(';'),
-                ],
-            ];
+        yield 'relative 1' => [
+            [
+                new Token([T_OPEN_TAG, "<?php\n"]),
+                new Token([T_NAMESPACE, 'namespace']),
+                new Token([T_NS_SEPARATOR, '\\']),
+                new Token([T_STRING, 'Transformer']),
+                new Token(';'),
+            ],
+            [
+                new Token([T_OPEN_TAG, "<?php\n"]),
+                new Token([T_NAME_RELATIVE, 'namespace\Transformer']),
```

This currently emits:
```
Warning: Use of undefined constant T_NAME_RELATIVE - assumed 'T_NAME_RELATIVE' (this will throw an Error in a future version of PHP) in C:\forks\PHP-CS-Fixer\tests\Tokenizer\Transformer\NameQualifiedTransformerTest.php on line 88
```
when run on PHP 7.4. So a condition is needed here.

---

How do you get this? I cannot see it (latest `master`):
```cli
$ ./vendor/bin/phpunit tests/Tokenizer/Transformer/NameQualifiedTransformerTest.php -vvv
PHPUnit 9.6.23 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.4.33
Configuration: /home/kuba/code/friendsofphp/PHP-CS-Fixer/phpunit.xml.dist

  Warning - The configuration file did not pass validation!
  The following problems have been detected:

  Line 20:
  - Element 'phpunit', attribute 'cacheDirectory': The attribute 'cacheDirectory' is not allowed.

  Line 44:
  - Element 'source': This element is not expected.

  Test results may not be as expected.


SSSSSSSSS

Time: 00:00.006, Memory: 6.00 MB

There were 9 skipped tests:

1) PhpCsFixer\Tests\Tokenizer\Transformer\NameQualifiedTransformerTest::testGetPriority
__OFFSET_LINE=26
__OFFSET_FILE=/home/kuba/code/friendsofphp/PHP-CS-Fixer/vendor/phpunit/phpunit/src/Framework/TestCase.php
PHP >= 8.0 is required.

phpvfscomposer:///home/kuba/code/friendsofphp/PHP-CS-Fixer/vendor/phpunit/phpunit/phpunit:106

2) PhpCsFixer\Tests\Tokenizer\Transformer\NameQualifiedTransformerTest::testGetName
__OFFSET_LINE=26
__OFFSET_FILE=/home/kuba/code/friendsofphp/PHP-CS-Fixer/vendor/phpunit/phpunit/src/Framework/TestCase.php
PHP >= 8.0 is required.

phpvfscomposer:///home/kuba/code/friendsofphp/PHP-CS-Fixer/vendor/phpunit/phpunit/phpunit:106

3) PhpCsFixer\Tests\Tokenizer\Transformer\NameQualifiedTransformerTest::testGetCustomTokens
__OFFSET_LINE=26
__OFFSET_FILE=/home/kuba/code/friendsofphp/PHP-CS-Fixer/vendor/phpunit/phpunit/src/Framework/TestCase.php
PHP >= 8.0 is required.

phpvfscomposer:///home/kuba/code/friendsofphp/PHP-CS-Fixer/vendor/phpunit/phpunit/phpunit:106

4) PhpCsFixer\Tests\Tokenizer\Transformer\NameQualifiedTransformerTest::testGetRequiredPhpVersionId
__OFFSET_LINE=26
__OFFSET_FILE=/home/kuba/code/friendsofphp/PHP-CS-Fixer/vendor/phpunit/phpunit/src/Framework/TestCase.php
PHP >= 8.0 is required.

phpvfscomposer:///home/kuba/code/friendsofphp/PHP-CS-Fixer/vendor/phpunit/phpunit/phpunit:106

5) PhpCsFixer\Tests\Tokenizer\Transformer\NameQualifiedTransformerTest::testTransformersIsFinal
__OFFSET_LINE=26
__OFFSET_FILE=/home/kuba/code/friendsofphp/PHP-CS-Fixer/vendor/phpunit/phpunit/src/Framework/TestCase.php
PHP >= 8.0 is required.

phpvfscomposer:///home/kuba/code/friendsofphp/PHP-CS-Fixer/vendor/phpunit/phpunit/phpunit:106

6) PhpCsFixer\Tests\Tokenizer\Transformer\NameQualifiedTransformerTest::testTransformDoesNotChangeSimpleCode
__OFFSET_LINE=26
__OFFSET_FILE=/home/kuba/code/friendsofphp/PHP-CS-Fixer/vendor/phpunit/phpunit/src/Framework/TestCase.php
PHP >= 8.0 is required.

phpvfscomposer:///home/kuba/code/friendsofphp/PHP-CS-Fixer/vendor/phpunit/phpunit/phpunit:106

7) PhpCsFixer\Tests\Tokenizer\Transformer\NameQualifiedTransformerTest::testNotDefiningConstructor
__OFFSET_LINE=26
__OFFSET_FILE=/home/kuba/code/friendsofphp/PHP-CS-Fixer/vendor/phpunit/phpunit/src/Framework/TestCase.php
PHP >= 8.0 is required.

phpvfscomposer:///home/kuba/code/friendsofphp/PHP-CS-Fixer/vendor/phpunit/phpunit/phpunit:106

8) PhpCsFixer\Tests\Tokenizer\Transformer\NameQualifiedTransformerTest::testProcess
__OFFSET_LINE=26
__OFFSET_FILE=/home/kuba/code/friendsofphp/PHP-CS-Fixer/vendor/phpunit/phpunit/src/Framework/TestCase.php
PHP >= 8.0 is required.

phpvfscomposer:///home/kuba/code/friendsofphp/PHP-CS-Fixer/vendor/phpunit/phpunit/phpunit:106

9) PhpCsFixer\Tests\Tokenizer\Transformer\NameQualifiedTransformerTest::testPriority
__OFFSET_LINE=26
__OFFSET_FILE=/home/kuba/code/friendsofphp/PHP-CS-Fixer/vendor/phpunit/phpunit/src/Framework/TestCase.php
PHP >= 8.0 is required.

phpvfscomposer:///home/kuba/code/friendsofphp/PHP-CS-Fixer/vendor/phpunit/phpunit/phpunit:106

OK, but incomplete, skipped, or risky tests!
Tests: 14, Assertions: 0, Skipped: 9.
```

---

It took me a while to reproduce it - you need to revert #8837.

repro: https://3v4l.org/rJNaT#veol

`XXX` - emits a warning in PHP 7.4
`\XXX` - emits an error/exception in PHP 7.4

But for some reasons, the error is swallowed by PHPUnit when skipped, but the data providers is evaluated even when there is `@requires PHP 8.0` above the class. So the constant/testcase should check if it is defined like before.

---

> So the constant/testcase should check if it is defined like before.

Can you raise PR?